article/sections/_6_graph_discovery.tex

```diff
@@ -0,0 +1,183 @@
+
+First, we start with a fully disconnected graph.
+To this graph, we add directed causal relationships to increase the score of the generative model that corresponds to our graph.
+Once we cannot improve the score this way anymore, we have found the graph structure that maximizes the score on our data.
+From this graph, we then prune as many directed causal relationships as possible without harming our score.
+In the end, the algorithms return the resulting set of graphs that retain maximal score.
+Common score-based algorithms include the Greedy Equivalence Search (GES) algorithm \citep{chickering_2002_optimal}.
+Typically these methods operate under stricter assumptions than constraint-based methods.
+Nonetheless, combinations of score and constraint-based ideas have outperformed methods from either class alone \citep{glymour_2019_review}.
+
+% DESCRIBE FCM ALGORITHMS
+Lastly, the third class of algorithms estimates Functional Causal Models (FCM) \citep{goudet_2018_learning} for each of the variables in our system.
+The defining characteristic in such algorithms is that they exploit asymmetries in the residuals of models for the hypothesized relationships of $X \rightarrow Y$, $Y \rightarrow X$.
+In particular, the residuals will be independent of the hypothesized cause in only one of the two potential models.
+As noted in the description of such algorithms, they are especially useful for discovering causal relationships between pairs of variables.
+In other words, discovery methods based on FCMs enable orientation of a graph's undirected or bi-directed edges.
+By orienting these edges, FCMs reduce ambiguity over whether $X \rightarrow Y$ or $Y \rightarrow X$.
+This orienting capability can be usefully combined with the previous causal discovery algorithms that infer classes graphs with equivalent marginal and conditional independencies.
+And of course, extensions of these techniques exist for inference in the presence of unobserved confounding \citep[Sec. 6]{goudet_2018_learning} and for learning an entire causal graph as opposed to dealing solely with variable pairs \citep{zheng_2020_learning}.
+
+\subsection{An application of causal discovery}
+\label{sec:discovery-application}
+To demonstrate the methods in this section, we chose to use the simplest causal discovery algorithm: the PC algorithm \citep{glymour_2001_causation}.
+For practitioners who may reasonably start with the simplest method they can find and increase methodological complexity as needed, this should be an illuminating starting point.
+Moreover, as noted above, the PC algorithm assumes that one has no unobserved confounders.
+As noted in Section \ref{sec:testing-demonstration}, we suspect that our causal graph for the drive alone utility of our dataset contains unobserved confounding.
+Accordingly, using the PC algorithm on our example allows us to observe how the algorithm behaves under violation of its assumptions.
```

Comment:
```suggestion
As noted in Section \ref{sec:testing-demonstration}, we suspect that our causal graph for the drive alone utility of our dataset contains unobserved confounding.
Given the PC algorithm assumes that one has no unobserved confounders, using it on our example allows us to observe how the algorithm behaves under violation of its assumptions.
```

Reply:
Done! See PR [here](https://github.com/hassan-obeid/tr_b_causal_2020/pull/155) as it'll be easier for me to merge.